### DISCUSSION TITLE
Character movement component

### DISCUSSION THREAD

**Discussion by ihavenick** [nxrighthere/UnrealCLR#98] [Help] [answered]
How can i get it in C# ?
It seems its missing in API

**Comment by nxrighthere** [comment, marked as answer]
This API is not exposed yet. Please, see the [roadmap](https://github.com/users/nxrighthere/projects/5?fullscreen=true) (it's movement components API).

**Reply by ihavenick** [reply]
I was look it but doesnt realized it.
Thank you.

**Reply by nxrighthere** [reply]
I'll give this task a higher priority.

**Reply by nxrighthere** [reply]
The movement component and rotating movement component exposed in https://github.com/nxrighthere/UnrealCLR/commit/d80cc19f843a8b9061bdbf01385930c278b8b892 and https://github.com/nxrighthere/UnrealCLR/commit/8a43fd1fa14d5cb0f5bf3a0e8106c1da71a7eb0a.

What [child components](https://docs.unrealengine.com/4.27/en-US/API/Runtime/Engine/GameFramework/UMovementComponent/) and relevant functionality you would like to see exposed?

**Reply by ihavenick** [reply]
Thats enough for me. Thank you !

**Reply by ihavenick** [reply]
Soo i was wrong. I cant get Character Movement Component with


 `rp.Velocity = Test.GetComponent<MovementComponent>().GetVelocity();`

**Reply by nxrighthere** [reply]
`MovementComponent` is an abstract class, you have to use `RotatingMovementComponent` at least. Please, let me know if you are interested in other [child components](https://docs.unrealengine.com/4.27/en-US/API/Runtime/Engine/GameFramework/UMovementComponent/).

**Reply by ihavenick** [reply]
Can't get CharacterMovement component with GetComponent<RotatingMovementComponent>. 
With name and without name.

This is my test code if you are interested https://github.com/ihavenick/PKSIW-CSHARP/blob/master/Rewind.cs

**Reply by nxrighthere** [reply]
Sorry, [UCharacterMovementComponent](https://docs.unrealengine.com/4.27/en-US/API/Runtime/Engine/GameFramework/UCharacterMovementComponent/) is missed at the moment. I'll take care of this.
